libbzip2-rs-sys/src/decompress.rs

```diff
@@ -261,21 +263,12 @@ pub(crate) fn decompress(strm: &mut bz_stream, s: &mut DState) -> ReturnCode {
                         break;
                     }
 
-                    if $strm.avail_in == 0 {
+                    if let Some(next_byte) = strm.read_byte() {
+                        $s.bsBuff = $s.bsBuff << 8 | next_byte as u32;
+                        $s.bsLive += 8;
```

Comment:
as a note, I suspect speedup when we make the bsBuff a u64 and/or try to perform wider reads here.